index.html

```diff
@@ -4975,6 +4975,646 @@ <h2 id="AudioWorkerParamDescriptor">
+          </dt>
+          <dd>
+            <p>
+              Imports a module script at <code>moduleURL</code>. The returned
+              promise is resolved when the code is successfully imported and
+              parsed. It is rejected if any uncaught exception occurs during
+              the execution of the imported script.
+            </p>
+          </dd>
+        </dl>
+        <section>
+          <h2 id="AudioWorkletNode">
+            The <dfn>AudioWorkletNode</dfn> interface
+          </h2>
+          <p>
+            This interface represents a user-defined
+            <a><code>AudioNode</code></a> which lives on the main <a>control
+            thread</a>. The user can create an <code>AudioWorkletNode</code>
+            from an <a><code>AudioContext</code></a> and such a node can be
+            connected with other built-in <a><code>AudioNode</code></a>s to
+            form an audio graph.
+          </p>
+          <dl title="interface AudioWorkletNode : AudioNode" class="idl">
+            <dt>
+              Constructor(BaseAudioContext context, DOMString name, optional
+              AudioWorkletNodeOptions options)
+            </dt>
+            <dd>
+              <p>
+                The construction procedure of an
+                <a><code>AudioWorkletNode</code></a> and the corresponding
+                <a><code>AudioWorkletProcessor</code></a> object consists of
+                several coordinated interactions. The following algorithm
+                specifies the steps in constructing an
+                <a><code>AudioWorkletNode</code></a> with the given parameters.
+                This sequence must be performed by the user agent <a data-lt=
+                "atomic">atomically</a>.
+              </p>
+              <ol>
+                <li>
+                  <em>In the <code>AudioWorkletGlobalScope</code></em>: Locate
+                  a definition of <a><code>AudioWorkletProcessor</code></a>
+                  subclass that is registered under the <code>name</code>
+                  argument. If none is found, an InvalidAccessError exception
+                  MUST be thrown by the <a><code>AudioWorkletNode</code></a>
+                  constructor.
+                </li>
+                <li>Run the <a><code>AudioWorkletProcessor</code></a>
+                constructor, passing a structured clone of the
+                <code>options</code> parameter that was provided to
+                <a><code>AudioWorkletNode</code></a>.
+                  <ol>
+                    <li>Run the constructor of the
+                    <a><code>AudioWorkletProcessor</code> subclass</a>. In this
+                    constructor, The <code>options</code> parameter can be used
+                    to initialize the state of the processor.
+                    </li>
+                  </ol>
+                </li>
+                <li>
+                  <em>In the <code>window</code> scope</em>: Construct
```

Comment:
Are we actually going to support WebAudio in a worker?  I thought we
weren't, especially in v1.

On Fri, Jul 22, 2016 at 7:47 AM, Paul Adenot <notifications@github.com>
wrote:

> In index.html
> <https://github.com/WebAudio/web-audio-api/pull/869#discussion_r71890606>:
>
> > +                  constructor.
> > +                </li>
> > +                <li>Run the <a><code>AudioWorkletProcessor</code></a>
> > +                constructor, passing a structured clone of the
> > +                <code>options</code> parameter that was provided to
> > +                <a><code>AudioWorkletNode</code></a>.
> > +                  <ol>
> > +                    <li>Run the constructor of the
> > +                    <a><code>AudioWorkletProcessor</code> subclass</a>. In this
> > +                    constructor, The <code>options</code> parameter can be used
> > +                    to initialize the state of the processor.
> > +                    </li>
> > +                  </ol>
> > +                </li>
> > +                <li>
> > +                  <em>In the <code>window</code> scope</em>: Construct
>
> We should use something more generic than window here. At some point,
> we'll want to initialize AudioContexts from a worker thread, and there is
> no window there. Maybe just reference the Global where it all started?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/WebAudio/web-audio-api/pull/869/files/b90832134ec0ca31b591dc5af332a76912a799de#r71890606>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAofPCiNc0UVoHNrKeSgl6M6QS4xm6Gcks5qYNfngaJpZM4JBSo7>
> .
>



-- 
Ray